pkg/vcs/vcs.go

```diff
@@ -72,6 +72,10 @@ type Repo interface {
 
 	// CommitExists check for the commit presence in local checkout.
 	CommitExists(commit string) (bool, error)
+
+	// PushCurrentCommitAsTagTo is used to cache commits elsewhere.
+	// Returns nil error is called with empty string.
+	PushCurrentCommitAsTagTo(string, ...string) error
```

Comment:
Nit: maybe pick some simpler name? `ArchiveCommit` or `PushCommit`. It's unlikely we'll have many git push methods.

Reply:
ArchiveCommit - Ok. I like long names because the help to understand code later.

Reply:
Renamed everything to "archive". It looks better than "cache".